hbase-common/src/main/java/org/apache/hadoop/hbase/io/TimeRange.java

```diff
@@ -18,24 +18,23 @@
 
 package org.apache.hadoop.hbase.io;
 
-import org.apache.hadoop.hbase.util.Bytes;
 import org.apache.yetus.audience.InterfaceAudience;
 
 /**
  * Represents an interval of version timestamps. Presumes timestamps between
  * {@link #INITIAL_MIN_TIMESTAMP} and {@link #INITIAL_MAX_TIMESTAMP} only. Gets freaked out if
  * passed a timestamp that is < {@link #INITIAL_MIN_TIMESTAMP},
- * <p>
+ * <p/>
  * Evaluated according to minStamp &lt;= timestamp &lt; maxStamp or [minStamp,maxStamp) in interval
  * notation.
- * <p>
+ * <p/>
  * Can be returned and read by clients. Should not be directly created by clients. Thus, all
  * constructors are purposely @InterfaceAudience.Private.
- * <p>
+ * <p/>
  * Immutable. Thread-safe.
  */
 @InterfaceAudience.Public
-public class TimeRange {
+public final class TimeRange {
```

Comment:
Probably this needs to be called out separately in the release notes as it is not binary-compatible. Shouldn't matter as it is introduced in major release, but probably worth it.

Reply:
Will mention this in release note. We removed all the public constructors so it should be final...